File: gameeliza/biblioteca/python/game_loop.py
import time

FPS = 60
STEP = 1.0 / FPS
running = True
previous = time.perf_counter()
accumulator = 0.0

while running:
    now = time.perf_counter()
    accumulator += min(now - previous, 0.25)
    previous = now
    while accumulator >= STEP:
        # update(STEP)
        accumulator -= STEP
    # render(accumulator / STEP)
    time.sleep(0.001)

File: gameeliza/biblioteca/cpp/game_loop.cpp
#include <chrono>

int main() {
    using clock = std::chrono::steady_clock;
    constexpr double step = 1.0 / 60.0;
    auto previous = clock::now();
    double accumulator = 0.0;
    for (;;) {
        auto now = clock::now();
        accumulator += std::chrono::duration<double>(now - previous).count();
        previous = now;
        while (accumulator >= step) {
            // update(step);
            accumulator -= step;
        }
        // render();
    }
}
